lib/tasks/import.rake

```diff
@@ -44,10 +43,38 @@ namespace :import do
     json_from_lines(args.filename) do |legacy_dataset|
       counter += 1
       print "Completed #{counter}\r"
-      Legacy::DatasetImportService.new(legacy_dataset, orgs_cache, topic_cache).run
+      Legacy::DatasetImportService.new(legacy_dataset, organisation_cache, topic_cache).run
     end
     logger.info 'Import complete'
   end
+
+  desc "Import a single legacy dataset from the legacy API"
+  task :single_legacy_dataset, [:legacy_shortname] => :environment do |_, args|
+    logger = Rails.logger
+
+    begin
+      api_parameters = { params: { id: args.legacy_shortname } }
+      api_response = RestClient.get 'https://data.gov.uk/api/3/action/package_show', api_parameters
+    rescue RestClient::ExceptionWithResponse => e
+      logger.error "Request to API to retrieve #{args.legacy_shortname} responded with: #{e.response.code}"
+      next
+    end
+
+    legacy_dataset = JSON.parse(api_response).fetch('result')
+    Legacy::DatasetImportService.new(legacy_dataset, organisation_cache, topic_cache).run
+
+    indexer = Legacy::DatasetIndexService.new
+    indexer.remove_from_index(legacy_dataset['id'])
```

Comment:
Would it be better to create before we remove in case the latter goes wrong?

Reply:
I'm not sure, the rake task is only for a specific use (backfillint missing data) so as long as it is in the database, we can reindex anytime.  Plus, don't want to remove the one we just added by mistake.  Without removing first, we end up with two entries.

Reply:
OK. I was missing the context that we had a 'live backup' if it goes wrong. Duplication in the new stack is a more important issue - all good :-)